### PR TITLE
Move Webpack build to the same Maven module from which it is invoked

### PR DESCRIPTION
**Pull request by basil**
#9718 moved the Webpack build from the `war/` Maven module to the root Maven module, but it did not move the corresponding source files to the same module, resulting in a setup that is a bit difficult to read because the root module build reaches into a different module to grab its source files. This PR improves the situation by moving the source files closer to where they are invoked. The output files still go to `war/src/main/webapp/jsbundles` so that they can be picked up by the WAR packaging, which seems incorrect (a different Maven module, a path that has `src` in it for generated output, etc.), but that is a pre-existing issue that this PR isn't making any worse than it was before.

### Testing done

`mvn clean verify -Dtest=hudson.AboutJenkinsTest`; compared the WAR before and after this PR to ensure there were no regressions

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@mention

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -39,4 +39,4 @@
 
 # Yarn
 # https://yarnpkg.com/getting-started/qa#which-files-should-be-gitignored
-/war/.yarn/plugins/** binary
+/.yarn/plugins/** binary
```
Comment (basil) [inline .gitattributes:42]: Not related to this PR conceptually, but fixing a bug introduced in #9718 while I was here.

File: .prettierignore [hunk 2]
```diff
@@ -8,10 +8,10 @@ node/
 .git
 
 # libraries / external deps / generated files
-war/src/main/js/plugin-setup-wizard/bootstrap-detached.js
+src/main/js/plugin-setup-wizard/bootstrap-detached.js
```
Comment (basil) [inline .prettierignore:11]: Part of the Webpack build invoked from the root Maven module, so the source file moves to the root Maven module.

File: .prettierignore [hunk 3]
```diff
@@ -8,10 +8,10 @@ node/
...
 war/src/main/webapp/scripts/yui
 war/src/main/webapp/jsbundles/
-war/src/main/scss/_bootstrap.scss
+src/main/scss/_bootstrap.scss
```
Comment (basil) [inline .prettierignore:14]: Part of the Webpack build invoked from the root Maven module, so the source file moves to the root Maven module.

File: .stylelintrc.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
 module.exports = {
   extends: "stylelint-config-standard",
   customSyntax: "postcss-scss",
-  ignoreFiles: ["war/src/main/scss/_bootstrap.scss"],
+  ignoreFiles: ["src/main/scss/_bootstrap.scss"],
```
Comment (basil) [inline .stylelintrc.js:4]: Part of the Webpack build invoked from the root Maven module, so the source file moves to the root Maven module.

File: eslint.config.cjs [hunk 5]
```diff
@@ -18,7 +18,7 @@ module.exports = [
       // External scripts
       ".pnp.cjs",
       ".pnp.loader.mjs",
-      "war/src/main/js/plugin-setup-wizard/bootstrap-detached.js",
+      "src/main/js/plugin-setup-wizard/bootstrap-detached.js",
```
Comment (basil) [inline eslint.config.cjs:21]: Part of the Webpack build invoked from the root Maven module, so the source file moves to the root Maven module.

File: eslint.config.cjs [hunk 6]
```diff
@@ -91,8 +91,8 @@ module.exports = [
   {
     files: [
       "eslint.config.cjs",
-      "war/postcss.config.js",
-      "war/webpack.config.js",
+      "postcss.config.js",
+      "webpack.config.js",
```
Comment (basil) [inline eslint.config.cjs:95]: Part of the Webpack build invoked from the root Maven module, so the build definition moves to the root Maven module.

File: package.json [hunk 7]
```diff
@@ -10,16 +10,16 @@
   },
   "private": true,
   "scripts": {
-    "dev": "webpack --config war/webpack.config.js",
-    "prod": "webpack --config war/webpack.config.js --mode=production",
+    "dev": "webpack --config webpack.config.js",
+    "prod": "webpack --config webpack.config.js --mode=production",
```
Comment (basil) [inline package.json:14]: Part of the Webpack build invoked from the root Maven module, so the build definition moves to the root Maven module.

File: package.json [hunk 8]
```diff
@@ -10,16 +10,16 @@
...
     "build": "yarn prod",
     "start": "yarn dev --watch",
     "lint:js": "eslint . && prettier --check .",
     "lint:js-ci": "eslint . -f checkstyle -o target/eslint-warnings.xml && prettier --check .",
-    "lint:css": "stylelint war/src/main/scss",
-    "lint:css-ci": "stylelint war/src/main/scss --custom-formatter stylelint-checkstyle-reporter -o target/stylelint-warnings.xml",
+    "lint:css": "stylelint src/main/scss",
+    "lint:css-ci": "stylelint src/main/scss --custom-formatter stylelint-checkstyle-reporter -o target/stylelint-warnings.xml",
```
Comment (basil) [inline package.json:20]: Part of the Webpack build invoked from the root Maven module, so the source file moves to the root Maven module.

File: package.json [hunk 9]
```diff
@@ -10,16 +10,16 @@
...
     "lint:ci": "yarn lint:js-ci && yarn lint:css-ci",
-    "lint:fix": "eslint --fix . && prettier --write . && stylelint war/src/main/scss --fix",
+    "lint:fix": "eslint --fix . && prettier --write . && stylelint src/main/scss --fix",
```
Comment (basil) [inline package.json:22]: Part of the Webpack build invoked from the root Maven module, so the source file moves to the root Maven module.

File: webpack.config.js [hunk 10]
```diff
@@ -57,7 +57,7 @@ module.exports = (env, argv) => ({
     styles: [path.join(__dirname, "src/main/scss/styles.scss")],
   },
   output: {
-    path: path.join(__dirname, "src/main/webapp/jsbundles"),
+    path: path.join(__dirname, "war/src/main/webapp/jsbundles"),
```
Comment (basil) [inline webpack.config.js:60]: This is still messy, but no worse than it was before this PR. Ideally they would be produced in a `target/` directory (ideally in the root project, where the sources now live) and fetched from there in the WAR module's configuration, but that is a project for another day.

**Review by timja — APPROVED** [review]

**Comment by basil** [comment]
_De minimis_

**Pull request merged**